src/com/cloudogu/ces/cesbuildlib/Git.groovy

```diff
@@ -113,6 +143,25 @@ class Git implements Serializable {
         return repositoryName
     }
 
+    /**
+     * Creates a release on Github and fills it with the changes provided
+     */
+    void addGithubRelease(String releaseVersion, String changes){
+        def repositoryName = getRepositoryName()
+        if (credentials) {
+            script.withCredentials([script.usernamePassword(credentialsId: credentials, usernameVariable: 'GIT_AUTH_USR', passwordVariable: 'GIT_AUTH_PSW')]) {
+                def body = "'{\"tag_name\": \"${releaseVersion}\", \"target_commitish\": \"master\", \"name\": \"${releaseVersion}\", \"body\":\"${changes}\"}'"
+                def apiUrl = "https://api.github.com/repos/${repositoryName}/releases"
+                def flags = "--request POST --data ${body} --header \"Content-Type: application/json\""
+                def username='\$GIT_AUTH_USR'
+                def password='\$GIT_AUTH_PSW'
+                script.sh "curl -u ${username}:${password} ${flags} ${apiUrl}"
+            }
+        } else {
+            throw new Exception("Unable to create Github release without credentials")
+        }
+    }
```

Comment:
+1. I know there already is `pushGitHubPagesBranch()`. Implementing it in git wasn't a good choice because it mixes Git and GitHub. This is a good time to refactor, before the git class gets too bloated.  So please move pushGitHubPagesBranch()` it to new `GitHub` class as well.
But please leave `pushGitHubPagesBranch()` in `Git` with `@Deprecated` annotation to retain API compatibility for now. I think `((GitHub)this).pushGitHubPagesBranch()` could work, don't you think?

Reply:
I moved this method and pushGitHubPagesBranch() to a new GitHub class. pushGitHubPagesBranch is still in Git class but marked as deprecated.